mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBSessionCLIQueryTest.java

```diff
@@ -1245,4 +1248,46 @@ public void testMapReduceWithOptions() throws Exception {
 		}
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "OGM-1433")
+	public void testDropCollection() {
+		try ( OgmSession session = openSession() ) {
+			Transaction transaction = session.beginTransaction();
+			final MarkTwainPoem genius = new MarkTwainPoem( 1L, "Genius", "Mark Twain" );
+			session.persist( genius );
+			session.flush();
+			String nativeQuery = "db." + MarkTwainPoem.TABLE_NAME + ".drop()";
+			int result = session.createNativeQuery( nativeQuery ).executeUpdate();
+			assertThat( result ).isEqualTo( 1 );
```

Comment:
Can you also add a check to make sure that the collection doesn't exist anymore on the db? 
Obtaining the list of collections should look something like this: 
```
		MongoDBDatastoreProvider provider = MongoDBTestHelper.getProvider( sessionFactory );
		MongoIterable<String> listCollectionNames = provider.getDatabase().listCollectionNames();
```

Reply:
Taken Care.

Reply:
Cool. thanks